File: C++/SWEA/D3/1289. 원재의 메모리 복구하기/원재의 메모리 복구하기.cpp
#include<iostream>

using namespace std;

int main() {
    int T;

    cin >> T;

    for (int test_case = 1; test_case <= T; ++test_case) {
        string origin;
        int ans = 0;
        char temp = '0';

        cin >> origin;

        for (char c: origin) {
            if (c == temp) continue;

            temp = c;
            ans++;
        }

        cout << '#' << test_case << ' ' << ans << '\n';
    }

    return 0;
}

File: Java/SWEA/D3/1289. 원재의 메모리 복구하기/원재의 메모리 복구하기.java
import java.util.*;
import java.io.*;
 
class Solution
{
    public static void main(String args[]) throws Exception
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
         
        for(int tc = 1; tc <= T; tc++)
        {
            int ans = 0;
            char temp = '0';
            String origin = br.readLine();
             
            for (char c : origin.toCharArray()) {
                if (c == temp)
                    continue;
                temp = (temp == '0' ? '1' : '0');
                ans++;
            }
            System.out.println("#" + tc + " " + ans);
        }
    }
}